scrape_books/spiders/books.py

```diff
@@ -0,0 +1,47 @@
+import scrapy
+
+from scrapy.http import Response
+from typing import Any
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs: Any) -> None:
+        for book in response.css("article.product_pod"):
+            detail_url = response.urljoin(book.css("a::attr(href)").get())
+            yield scrapy.Request(
+                url=detail_url,
+                callback=self._parse_book_detail
+            )
+
+        next_page = response.css("li.next a::attr(href)").get()
+        if next_page is not None:
+            yield response.follow(next_page, callback=self.parse)
+
+    def _parse_book_detail(self, response: Response) -> dict[str, Any]:
+        yield {
+            "title": response.css(
+                ".product_main > h1::text"
+            ).get(),
+            "price": float(response.css(
+                ".product_main > .price_color::text"
+            ).get().replace("£", "")),
+            "amount_in_stock": int(response.css(
+                ".product_main > p.instock.availability::text"
+            ).getall()[1].replace("\n", "").split()[2].replace("(", "")),
```

Comment:
The `amount_in_stock` extraction assumes that the stock information is always in the same format and position. Ensure that `getall()[1]` is the correct index for the stock number, as this might lead to an `IndexError` if the format changes or if there are fewer elements than expected.